integration/spark/src/main/scala/org/apache/carbondata/view/MVManagerInSpark.scala

```diff
@@ -48,17 +48,14 @@ class MVManagerInSpark(session: SparkSession) extends MVManager {
 
 object MVManagerInSpark {
 
-  private val MANAGER_MAP_BY_SESSION =
-    new util.HashMap[SparkSession, MVManagerInSpark]()
+  private var viewManager: MVManagerInSpark = null
 
+  // returns single MVManager instance for all the current sessions.
```

Comment:
Probably MVManagerInSpark instance is maintained per SparkSession to support multi-tenency.  Each user may have an independent and isolated set of databases/tables/conf etc sharing same sparkcontext. I think, we still need this isolation. right ?

Reply:
There should be only one MVManagerInSpark instance sharing the same sparkcontext. If not, in different spark sessions, the query will not hit MV.